src/rpp/rpp/schedulers/trampoline_scheduler.hpp

```diff
@@ -0,0 +1,117 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+//                            TC Wang 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/schedulers/fwd.hpp>                       // own forwarding
+#include <rpp/schedulers/details/worker.hpp>            // worker
+#include <rpp/subscriptions/composite_subscription.hpp> // lifetime
+#include <rpp/schedulers/details/queue_worker_state.hpp>// state
+
+#include <concepts>
+#include <chrono>
+#include <functional>
+#include <thread>
+
+
+namespace rpp::schedulers
+{
+
+template <typename F>
+struct scope_guard {
+    explicit scope_guard(F&& f) : m_f(std::move<F>(std::forward<F>(f))) {}
+    ~scope_guard() noexcept { m_f(); }
+    F m_f;
+};
+
+/**
+ * \brief schedules execution of schedulables via queueing tasks to the caller thread with priority to time_point and order
+ *
+ * \par Example
+ * \snippet trampoline.cpp trampoline
+ *
+ * \ingroup schedulers
+ */
+class trampoline final : public details::scheduler_tag
+{
+public:
+    class worker_strategy
+    {
+    public:
+        explicit worker_strategy(const rpp::composite_subscription& subscription)
+            : m_sub{subscription} {};
+
+        void defer_at(time_point time_point, std::invocable auto&& fn) const
+        {
+            if (!m_sub.is_subscribed())
+                return;
+
+            with_thread_local_schedulable_queue([&](auto& queue, auto& queue_in_use)
```

Comment:
It would be more understandable =)
```suggestion
            with_thread_local_schedulable_queue([&](auto& queue, bool& queue_in_use)
```